src/TagList/test/TagList.spec.js

```diff
@@ -24,6 +24,91 @@ describe(TagList.displayName, () => {
     expect(await driver.exists()).toBe(true);
   });
 
+  it('should not show toggle more button', async () => {
+    const { driver } = render(
+      <TagList
+        tags={[
+          {
+            id: '1',
+            children: 'Some Tag',
+          },
+        ]}
+      />,
+    );
+
+    expect(await driver.toggleMoreButtonExists()).toBe(false);
+  });
+
+  it('should show toggle more button when prop is passed', async () => {
+    const { driver } = render(
+      <TagList
+        tags={[
+          {
+            id: '1',
+            children: 'Some Tag',
+          },
+          {
+            id: '2',
+            children: 'Tag 2',
+          },
+          {
+            id: '3',
+            children: 'Tag 3',
+          },
+          {
+            id: '4',
+            children: 'Tag 4',
+          },
+        ]}
+        toggleMoreButton={(amountOfHiddenTags, isExpanded) => ({
+          label: isExpanded ? 'Show Less' : `+${amountOfHiddenTags} More`,
+          tooltipProps: !isExpanded
+            ? {
+                content: 'Show More',
+              }
+            : undefined,
+        })}
+      />,
+    );
+
+    expect(await driver.toggleMoreButtonExists()).toBe(true);
+  });
+
+  it('should click toggle more button', async () => {
+    const { driver } = render(
+      <TagList
+        tags={[
+          {
+            id: '1',
+            children: 'Some Tag',
+          },
+          {
+            id: '2',
+            children: 'Tag 2',
+          },
+          {
+            id: '3',
+            children: 'Tag 3',
+          },
+          {
+            id: '4',
+            children: 'Tag 4',
+          },
+        ]}
+        toggleMoreButton={(amountOfHiddenTags, isExpanded) => ({
+          label: isExpanded ? 'Show Less' : `+${amountOfHiddenTags} More`,
+          tooltipProps: !isExpanded
+            ? {
+                content: 'Show More',
+              }
+            : undefined,
+        })}
+      />,
+    );
+    await driver.clickToggleMoreButton();
```

Comment:
When it comes to testing the result of an action (like a click or hover) I like to test both the before and after states.

```suggestion
    expect(await driver.toggleMoreButtonLabel()).toEqual('+1 More');
    await driver.clickToggleMoreButton();
```